### DISCUSSION TITLE
AZERTY keyboard not working (Fedora 40 QUERTY -> Windows AZERTY)

### DISCUSSION THREAD

**Discussion by eric-saintetienne** [rustdesk/rustdesk#8920] [General]
### Bug Description

When typing on the local client (linux, QUERTY) the keys are not converted into AZERTY on the remote Windows rustdesk.

For example typing `q` on my local keyboard (linux) yields `a` on the remote client (Window)

### How to Reproduce

Connect from a Linux machine that has a QWERTY keyboard and layout to a Windows machine that has an AZERTY keyboard an layout.

### Expected Behavior

typing `q` should result in `q` (not `a`)

### Operating system(s) on local side and remote side

Fedora 40 (QWERTY) -> Windows 10 Pro (AZERTY)

Local client is latest `1.2.7` (rpm) on Fedora 40, using a British QWERTY keyboard and layout
Remote server is latest `1.2.7` (msi) on Windows 10 Pro (version 22H2), using an AZERTY keyboard and layout

### RustDesk Version(s) on local side and remote side

local: 1.2.7 (rpm) -> remote: 1.2.7 (msi)

(upgraded from 1.2.3 on which the issue was first seen)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/FAQ#keyboard-translation-modes

**Reply by eric-saintetienne** [reply]
Indeed map mode is active. I'll try with translate mode.

**Reply by eric-saintetienne** [reply]
> https://github.com/rustdesk/rustdesk/wiki/FAQ#keyboard-translation-modes

There's no such option as `Translate mode` in RustDesk 1.2.7:

![map-mode](https://github.com/user-attachments/assets/7210e883-4c26-40fe-951b-e28c8e4912cc)

When I click on `Map mode` nothing happens.

**Reply by fufesou** [reply]
Yes. Only "Map mode" is supported on Wayland for now.

You need to set input method  on the controlled side, eg. `QWERTY`.

**Reply by eric-saintetienne** [reply]
> Yes. Only "Map mode" is supported on Wayland for now.

I've installed X11 and indeed `Tranlate mode` is present and works as expected.
 
> You need to set input method on the controlled side, eg. `QWERTY`.

Would it be possible to instead support `Translated mode` on wayland instead? Thanks.

**Reply by fufesou** [reply]
It depends on RustDesk's plan.
I have not received any plans for this in the short term.
